common/test/acceptance/tests/studio/test_studio_settings.py

```diff
@@ -624,3 +624,281 @@ def test_upload_course_video_thumbnail_image(self):
+                        self.ENROLLMENT_START_DATE_CSS, self.ENROLLMENT_END_DATE_CSS]
+
+        expected_values = [self.course_start_date, '', '', '']
+        # Assert changes have been persistent.
+        self.assert_values(css_selector, expected_values)
+
+    def test_cannot_clear_the_course_start_date(self):
+        """
+        Scenario: User cannot clear the course start date
+        Given I have set course dates
+        And I press the "Save" notification button
+        And I clear the course start date
+        Then I receive a warning about course start date
+        And I reload the page
+        And the previously set start date is shown
+        """
+        # Set dates
+        self.set_course_dates()
+        # Save changes
+        self.settings_page.save_changes()
+        # Get default start date
+        default_start_date = self.settings_page.get_value(self.COURSE_START_DATE_CSS)
+        # Set course start date to empty
+        self.settings_page.set_value(self.COURSE_START_DATE_CSS, '')
+        # Make sure error message is show with appropriate message
+        error_message_css = '.message-error'
+        self.settings_page.wait_for_element_presence(error_message_css, 'Error message is present')
+        error_message = self.settings_page.get_text(error_message_css)
+        self.assertEquals(error_message, "The course must have an assigned start date.")
+
+        # Refresh the page and assert start date has not changed.
+        self.settings_page.refresh_and_wait_for_load()
+        self.assert_values([self.COURSE_START_DATE_CSS], [default_start_date])
+
+    def test_user_can_correct_course_start_date_warning(self):
+        """
+        Scenario: User can correct the course start date warning
+        Given I have tried to clear the course start
+        And I have entered a new course start date
+        And I press the "Save" notification button
+        Then The warning about course start date goes away
+        And I reload the page
+        Then my new course start date is shown
+        """
+        # Set course start date to empty
+        self.settings_page.set_value(self.COURSE_START_DATE_CSS, '')
+        # Make sure we get error message
+        error_message_css = '.message-error'
+        self.settings_page.wait_for_element_presence(error_message_css, 'Error message is present')
+        error_message = self.settings_page.get_text(error_message_css)
+        self.assertEquals(error_message, "The course must have an assigned start date.")
+        # Set new course start value
+        self.settings_page.set_value(self.COURSE_START_DATE_CSS, self.course_start_date)
+        self.settings_page.un_focus_input_field()
+        # Error message disappears
+        self.settings_page.wait_for_element_absence(error_message_css, 'Error message is not present')
+        # Save the changes and refresh the page.
+        self.settings_page.save_changes()
+        self.settings_page.refresh_and_wait_for_load()
```

Comment:
We are refreshing the page just to make sure changes are persistent/not persistent based upon the nature of the test when we reload the page. Also I am trying to imitate lettuce scenarios as is.